src/Codec/Xlsx/StyleSheet.hs

```diff
@@ -0,0 +1,1123 @@
+{-# LANGUAGE CPP                #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE RecordWildCards    #-}
+{-# LANGUAGE TemplateHaskell    #-}
+{-# OPTIONS_GHC -Wall #-}
+-- | Support for writing (but not reading) style sheets
+module Codec.Xlsx.StyleSheet (
+    -- * The main two types
+    StyleSheet(..)
+  , CellXf(..)
+  , minimalStyleSheet
+    -- * Supporting record types
+  , Alignment(..)
+  , Border(..)
+  , BorderStyle(..)
+  , Color(..)
+  , Fill(..)
+  , FillPattern(..)
+  , Font(..)
+  , Protection(..)
+    -- * Supporting enumerations
+  , CellHorizontalAlignment(..)
+  , CellVerticalAlignment(..)
+  , FontFamily(..)
+  , FontScheme(..)
+  , FontVerticalAlignment(..)
+  , LineStyle(..)
+  , PatternType(..)
+  , ReadingOrder(..)
+    -- * Lenses
+    -- ** StyleSheet
+  , styleSheetBorders
+  , styleSheetFonts
+  , styleSheetFills
+  , styleSheetCellXfs
+    -- ** CellXf
+  , cellXfApplyAlignment
+  , cellXfApplyBorder
+  , cellXfApplyFill
+  , cellXfApplyFont
+  , cellXfApplyNumberFormat
+  , cellXfApplyProtection
+  , cellXfBorderId
+  , cellXfFillId
+  , cellXfFontId
+  , cellXfNumFmtId
+  , cellXfPivotButton
+  , cellXfQuotePrefix
+  , cellXfId
+  , cellXfAlignment
+  , cellXfProtection
+    -- ** Alignment
+  , alignmentHorizontal
+  , alignmentIndent
+  , alignmentJustifyLastLine
+  , alignmentReadingOrder
+  , alignmentRelativeIndent
+  , alignmentShrinkToFit
+  , alignmentTextRotation
+  , alignmentVertical
+  , alignmentWrapText
+    -- ** Border
+  , borderDiagonalDown
+  , borderDiagonalUp
+  , borderOutline
+  , borderBottom
+  , borderDiagonal
+  , borderEnd
+  , borderHorizontal
+  , borderStart
+  , borderTop
+  , borderVertical
+  , borderLeft
+  , borderRight
+    -- ** BorderStyle
+  , borderStyleColor
+  , borderStyleLine
+    -- ** Color
+  , colorAutomatic
+  , colorARGB
+  , colorTheme
+  , colorTint
+    -- ** Fill
+  , fillPattern
+    -- ** FillPattern
+  , fillPatternBgColor
+  , fillPatternFgColor
+  , fillPatternType
+    -- ** Font
+  , fontBold
+  , fontCharset
+  , fontColor
+  , fontCondense
+  , fontExtend
+  , fontFamily
+  , fontItalic
+  , fontName
+  , fontOutline
+  , fontScheme
+  , fontShadow
+  , fontStrikeThrough
+  , fontSize
+  , fontUnderline
+  , fontVertAlign
+    -- ** Protection
+  , protectionHidden
+  , protectionLocked
+  ) where
+
+import Control.Lens hiding ((.=))
+import Data.Default
+import Data.Maybe (catMaybes)
+import Text.XML
+import qualified Data.Map as Map
+
+#if !MIN_VERSION_base(4,8,0)
+import Control.Applicative
+#endif
+
+import Codec.Xlsx.Writer.Internal
+
+{-------------------------------------------------------------------------------
+  The main types
+-------------------------------------------------------------------------------}
+
+-- | StyleSheet for an XML document
+--
+-- Relevant parts of the EMCA standard (2nd edition, part 1,
```

Comment:
Oh, sure :) Go for it! If no code changes indeed nothing needs to be tested again :) 

In that case I think you can remove the leading and trailing borders completely, I think that's the only real change we have seen so far between the two versions of the spec? And in that particular case I ended up using left and right anyway. 

> On 28 Nov 2015, at 19:27, Kirill Zaborsky <notifications@github.com> wrote:
> 
> In src/Codec/Xlsx/StyleSheet.hs:
> 
> > +import Text.XML
> > +import qualified Data.Map as Map
> > +
> > +#if !MIN_VERSION_base(4,8,0)
> > +import Control.Applicative
> > +#endif
> > +
> > +import Codec.Xlsx.Writer.Internal
> > +
> > +{-------------------------------------------------------------------------------
> > +  The main types
> > +-------------------------------------------------------------------------------}
> > +
> > +-- | StyleSheet for an XML document
> > +--
> > +-- Relevant parts of the EMCA standard (2nd edition, part 1,
> I don't understand what do you want to test when all what's required is just to change the references in the comments and some words in them, I haven't found yet any part of the code which could require change.
> What do you think if I will createa PR against your branch with haddocks updated for the 4th edition?
> 
> —
> Reply to this email directly or view it on GitHub.
>